tests/jballerina-unit-test/src/test/resources/test-src/workers/workers.bal

```diff
@@ -667,6 +667,17 @@ public function testWorkerWithQuery() {
     assertEquals(4, sum);
 }
 
+function testBindingPatternVariablesInWorker() {
+    record {|string name;|} rec = {name: "name"};
+    [string] [colon] = [";"];
+    var {name} = rec;
+
+    worker A returns string {
+        return name + colon;
+    }
+    string _ = wait A;
+}
+
```

Comment:
It is better to add a test with destructuring nested records.
(example- destructuring B)
```
type A record {
     string a = "";
};

type B record {
    A a1;
    string a2 = "";
};
```

Reply:
Have added such a test [here](https://github.com/poorna2152/ballerina-lang/blob/fix_-41236/tests/jballerina-unit-test/src/test/resources/test-src/closures/closure.bal#L830).